### PR TITLE
feat: case sensitive sorter

### PR DESCRIPTION
**Pull request by vlaw**
I'm not sure if this is useful or not. But I guess having one more option is not a bad idea. 😄

### CODE REVIEW

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -252,7 +252,7 @@ Here is a list of the options available in the setup call:
 
 *nvim-tree.sort_by*
 - |sort_by|: changes how files within the same directory are sorted. can be
-  one of 'name' | 'modification_time'
+  one of 'name' | 'Name' (case sensitive) | 'modification_time'
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:255]: Great idea, but the value is too complex.

How about `name_case_sensitive`

File: lua/nvim-tree/explorer/sorters.lua [hunk 2]
```diff
@@ -116,8 +128,10 @@ function M.setup(opts)
   M.sort_by = opts.sort_by
   if M.sort_by == "modification_time" then
     M.node_comparator = M.node_comparator_modification_time
+  elseif M.sort_by == "Name" then
```
Comment (alex-courtis) [inline lua/nvim-tree/explorer/sorters.lua:131]: name should be the default; the else

**Review by alex-courtis — CHANGES_REQUESTED** [review]

File: doc/nvim-tree-lua.txt (hunk 1 above)
Comment (vlaw) [reply doc/nvim-tree-lua.txt:255]: The new option that I use is “Name”， which means case sensitive name sorter. 
'name' is still the default option, case insensitive name sorter.
I guess the doc part is misleading, need some updates.

File: lua/nvim-tree/explorer/sorters.lua (hunk 2 above)
Comment (vlaw) [reply lua/nvim-tree/explorer/sorters.lua:131]: nae is still the default. 'Name' is for case sensitive.

**Review by kyazdani42 — APPROVED** [review]

**Review by alex-courtis — APPROVED** [review]

**Pull request merged**